src/main/java/httpServer/RequestHandler.java

```diff
@@ -0,0 +1,20 @@
+package httpServer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+class RequestHandler {
+    private final BufferedReader input;
+
+    public RequestHandler(BufferedReader input) {
+        this.input = input;
+    }
+
+    String readResponse() {
```

Comment:
If this is a request, then shouldn't this be `readRequest`, but maybe it really is a `RequestReader`,  so the method could just be `read`.

Reply:
Thanks for spotting this